### ISSUE TITLE
[a11y]: The button gives a default `id="icon"` when you have `hasIconOnly` passed in

### ISSUE THREAD

**Issue by setupotnis** [carbon-design-system/carbon#16094]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.59.5

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker

### Description

When there are multiple buttons which have `hasIconOnly`, the IBM accessibility checker gives an error saying that `id="icon"` is already in use. 

It would be good if you could provide an easy way to pass in an id for the icon so that the user could make the icon have a unique id, fixing the error shown by the IBM accessibility checker 

### WCAG 2.1 Violation

4.1.1 due to duplicate values of type ID

### Reproduction/example

Watsonx -> prompt lab -> chat mode

### Steps to reproduce

Create a project after logging in to watsonx
Create a prompt lab asset
Navigate to chat mode, you'll see the following error:
`The <svg> element has the id "icon" that is already in use.`

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand.

**Comment by setupotnis** [comment]
Hi @tay1orjones, went to reproduce it, and realized the issue was happening because we had a custom icon rendering with an `id="icon"`. 

Will close the issue since there's no fix necessary.
